clients/js/src/signed_calls.ts

```diff
@@ -21,8 +18,68 @@ const DEFAULT_GAS_LIMIT = 30_000_000;
 const DEFAULT_VALUE = 0;
 const DEFAULT_DATA = '0x';
 const zeroAddress = () => `0x${'0'.repeat(40)}`;
+const MAX_SAFE_BIG_NUM = 0x1fffffffffffff - 1;
 
-export type Signer = Pick<EthersSigner, 'getTransactionCount' | 'getChainId'> &
+class SignedCallCacheManager {
+  public enabled = true;
```

Comment:
Is this something that mainstream dapps will use? How hard would it be for the dapps that needs this to clear the cache when required, or make a subclass of this?